src/hotspot/share/memory/arena.cpp

```diff
@@ -280,15 +273,19 @@ void* Arena::grow(size_t x, AllocFailType alloc_failmode) {
     return nullptr;
   }
 
-  Chunk *k = _chunk;            // Get filled-up chunk address
-  _chunk = new (alloc_failmode, len) Chunk(len);
+  Chunk* k = _chunk;            // Get filled-up chunk address
+  _chunk = ChunkPool::allocate_chunk(len, alloc_failmode);
```

Comment:
Odd that we would fatal out if init chunk cannot be allocated, and here we leave the OOM behavior to the caller. Is this ever called with anything else than exit_oom?

Reply:
Looking at the callers: No, we don't, but I think changing this API would go under a different PR.